awslambda/src/test/scala/akka/stream/alpakka/awslambda/scaladsl/AwsLambdaFlowSpec.scala

```diff
@@ -0,0 +1,99 @@
+import org.mockito.ArgumentMatchers.{any => mockitoAny, eq => mockitoEq}
+import org.mockito.Mockito._
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mockito.MockitoSugar
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class AwsLambdaFlowSpec
+    extends TestKit(ActorSystem("AwsLambdaFlowSpec"))
+    with WordSpecLike
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with ScalaFutures
+    with Matchers
+    with MockitoSugar {
+
+  implicit val mat = ActorMaterializer()
+
+  implicit val ec = system.dispatcher
+
+  implicit val awsLambdaClient = mock[AWSLambdaAsyncClient]
+
+  override protected def afterEach(): Unit = {
+    reset(awsLambdaClient)
+    verifyNoMoreInteractions(awsLambdaClient)
+  }
+
+  override protected def afterAll(): Unit =
+    Await.ready(system.terminate(), 5.seconds)
+
+  "AwsLambdaFlow" should {
+
+    val invokeRequest = new InvokeRequest().withFunctionName("test_function").withPayload("test_payload")
+    val invokeFailureRequest = new InvokeRequest().withFunctionName("failure_function").withPayload("test_payload")
+    val invokeResult = new InvokeResult()
+    val lambdaFlow = AwsLambdaFlow(1)
+
+    "call a single invoke request" in {
+
+      when(awsLambdaClient.invokeAsync(mockitoEq(invokeRequest),
+          mockitoAny[AsyncHandler[InvokeRequest, InvokeResult]]())).thenAnswer(new Answer[AnyRef] {
+        override def answer(invocation: InvocationOnMock): AnyRef = {
+          invocation.getArgument[AsyncHandler[InvokeRequest, InvokeResult]](1).onSuccess(invokeRequest, invokeResult)
+          new CompletableFuture
+        }
+      })
+
+      val (probe, future) = TestSource.probe[InvokeRequest].via(lambdaFlow).toMat(Sink.seq)(Keep.both).run()
+      probe.sendNext(invokeRequest)
+      probe.sendComplete()
+
+      future.map(_ shouldBe invokeResult)
+      verify(awsLambdaClient, times(1)).invokeAsync(mockitoEq(invokeRequest),
+        mockitoAny[AsyncHandler[InvokeRequest, InvokeResult]]())
+
+    }
+
+    "call with exception" in {
+
+      when(awsLambdaClient.invokeAsync(mockitoAny[InvokeRequest](),
+          mockitoAny[AsyncHandler[InvokeRequest, InvokeResult]]())).thenAnswer(new Answer[Future[InvokeResult]] {
+        override def answer(invocation: InvocationOnMock): Future[InvokeResult] = {
+          invocation
+            .getArgument[AsyncHandler[InvokeRequest, InvokeResult]](1)
+            .onError(new RuntimeException("Error in lambda"))
+          new CompletableFuture[InvokeResult]()
+        }
+      })
+
+      val (probe, future) = TestSource.probe[InvokeRequest].via(lambdaFlow).toMat(Sink.seq)(Keep.both).run()
+
+      probe.sendNext(invokeFailureRequest)
+      probe.sendComplete()
+
+      future.failed.map(_ shouldBe a[RuntimeException])
+
+    }
```

Comment:
So the async-impl basically just delegates blocking to a thread-pool? If yes, I would prefer to not use their async client. The simple reason is that the Akka side is strictly more flexible when it comes to dispatchers, especially since it allows fusing some stages into the actor handling the blocking IO.